x/subspaces/legacy/v5/store_test.go

```diff
@@ -0,0 +1,100 @@
+package v5_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/desmos-labs/desmos/v4/app"
+	"github.com/desmos-labs/desmos/v4/testutil/storetesting"
+	v5 "github.com/desmos-labs/desmos/v4/x/subspaces/legacy/v5"
+	"github.com/desmos-labs/desmos/v4/x/subspaces/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMigrateStore(t *testing.T) {
+	cdc, legacyAmino := app.MakeCodecs()
+
+	// Build all the necessary keys
+	keys := sdk.NewKVStoreKeys(types.StoreKey, authtypes.StoreKey, paramstypes.StoreKey)
+	tKeys := sdk.NewTransientStoreKeys(paramstypes.TStoreKey)
+	memKeys := sdk.NewMemoryStoreKeys(capabilitytypes.MemStoreKey)
+
+	// Build the x/auth keeper
+	paramsKeeper := paramskeeper.NewKeeper(
+		cdc,
+		legacyAmino,
+		keys[paramstypes.StoreKey],
+		tKeys[paramstypes.TStoreKey],
+	)
+	authKeeper := authkeeper.NewAccountKeeper(
+		cdc,
+		keys[authtypes.StoreKey],
+		paramsKeeper.Subspace(authtypes.ModuleName),
+		authtypes.ProtoBaseAccount,
+		app.GetMaccPerms(),
+	)
+
+	testCases := []struct {
+		name      string
+		store     func(ctx sdk.Context)
+		shouldErr bool
+		check     func(ctx sdk.Context)
+	}{
+		{
+			name: "subspace is migrated properly",
+			store: func(ctx sdk.Context) {
+				oldSubspace := types.NewSubspace(
+					1,
+					"name",
+					"description",
+					"cosmos10ya9y35qkf4puaklx5fs07sxfxqncx9usgsnz6",
+					"cosmos10ya9y35qkf4puaklx5fs07sxfxqncx9usgsnz6",
+					"cosmos10ya9y35qkf4puaklx5fs07sxfxqncx9usgsnz6",
+					time.Date(2023, 1, 11, 1, 1, 1, 1, time.UTC),
+				)
+				ctx.KVStore(keys[types.StoreKey]).Set(types.SubspaceStoreKey(1), cdc.MustMarshal(&oldSubspace))
+			},
+			check: func(ctx sdk.Context) {
+				var newSubspace types.Subspace
+				err := cdc.Unmarshal(ctx.KVStore(keys[types.StoreKey]).Get(types.SubspaceStoreKey(1)), &newSubspace)
+				require.NoError(t, err)
+				require.Equal(t, types.NewSubspace(
+					1,
+					"name",
+					"description",
+					"cosmos1cyjzgj9j7d2gdqk78pa0fgvfnlzradat97aek9",
+					"cosmos10ya9y35qkf4puaklx5fs07sxfxqncx9usgsnz6",
+					"cosmos10ya9y35qkf4puaklx5fs07sxfxqncx9usgsnz6",
+					time.Date(2023, 1, 11, 1, 1, 1, 1, time.UTC),
+				), newSubspace)
```

Comment:
We should remain text generated by current method, it would detect the change and make sure the migration always properly if `types.GetTreasuryAddress` or `authtypes.NewModuleAddress` are changed.